backend/app/api/api_v1/routers/documents.py

```diff
@@ -78,6 +79,55 @@
 documents_router = APIRouter()
 
 
+document_respoonses: Mapping[int, Mapping] = {
+    404: {"description": "Cannot find a document that matches the filter criteria."}
+}
+
+documents_router.add_api_route
+
+
+@documents_router.get(
+    "/documents",
+    response_model=List[DocumentBrowseResponse],
+    summary="Get a list of documents",
+    responses=document_respoonses,
+)
+async def document_browse(
+    country_code: Union[str, None] = None,
+    q: Union[str, None] = None,
+    db=Depends(get_db),
+    current_user=Depends(get_current_active_user),
+):
+    """Filter all documents"""
+    query = (
+        db.query(
+            Document.name,
+            Document.description,
+            Document.publication_ts,
+            Geography.display_value.label("country_name"),
+            Geography.value.label("country_code"),
+        )
+        .join(Geography, Document.geography_id == Geography.id)
+        .join(DocumentType, Document.type_id == DocumentType.id)
+    )
+
+    if country_code is not None:
+        query = query.filter(Geography.value == country_code)
+
+    def row_to_response(row):
+        return DocumentBrowseResponse(
+            name=row["name"],
+            description=row["description"],
+            country_code=row["country_code"],
+            country_name=row["country_name"],
+            publication_ts=row["publication_ts"],
+        )
+
+    found = query.all()
+
+    return [*map(row_to_response, found)]
```

Comment:
Might be a bit more pythonic as:

```suggestion
    return [row_to_response(row) for row in query.all()]
```

Reply:
if my question about `row_to_response` is correct... this could even be
```suggestion
    return [DocumentBrowseResponse(**row) for row in query.all()]
```

Reply:
I'll stop looking now :joy: